lib/domain/favorite_model.dart

```diff
@@ -1,135 +1,114 @@
-    }
-  }
-
-  void hiveBoxClear() async {
-    await box.clear();
-  }
-
-  bool checkingForChange(List<ShowHive> data) {
-    bool flag = true;
-    int count = 0;
-
-    if (data.length == box.length) {
-      int countLenght = data.length;
-      for (int i = 0; i < data.length; i++) {
-        final boxInfo = box.getAt(i);
-        if (data[i].id == boxInfo?.id) {
-          count++;
-        }
-      }
-      if (count == countLenght) {
-        flag = false;
-      }
-    }
-    return flag;
-  }
-
-  // int checkingForFavorite (int id){
-  //       int counter = 0;
-
-  //   for (int index = 0; index < box.length; index++) {
-  //     final showinfo = box.getAt(index);
-  //     var linkImage = showinfo?.image ?? '';
-  //     var showName = showinfo?.name;
-  //     var showLanguage = showinfo?.language;
-
-  //     if (name == showName && language == showLanguage && image == linkImage) {
-  //       counter++;
-  //     }
-  //   }
-  //   return counter;
-
-  // }
-
-  // final FirebaseFirestore firestore = FirebaseFirestore.instance;
-  // //final CollectionReference usersCollection = firestore.collection('shows');
-
-  // // void modelFilling() {
-  // //   final CollectionReference usersCollection = firestore.collection('shows');
-  // // }
-
-  // Future<void> getUsers() async {
-  //   final CollectionReference usersCollection = firestore.collection('shows');
-  //   try {
-  //     final QuerySnapshot querySnapshot = await usersCollection.get();
-
-  //     final List<FavoriteModel> favoriteModel = querySnapshot.docs
-  //         .map((doc) =>
-  //             FavoriteModel.fromMap(doc.data() as Map<String, dynamic>))
-  //         .toList();
-  //     this.favoriteModel = favoriteModel;
-  //   } catch (e) {
-  //     print(e);
-  //   }
-  //   // int? lenght = favoriteModel.length ?? 0;
-  //   for (int index = 0;
-  //       index < 5;
-  //       //favoriteModel!.length;
-  //       index++) {
-  //     final fml = favoriteModel?[index];
-
-  //     print('${fml?.id} + ${fml?.name} + ${fml?.language} + ${fml?.image}');
-}
-//   }
-// }
+import 'dart:core';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:hive/hive.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:flutter_application_1/domain/hive_model.dart';
+
+class FirecloudeModel {
+  final favoriteLocalBox = Hive.box<ShowHive>('favoriteLocalBox');
+  final filterBox = Hive.box<ShowHive>('filterBox');
+
+  Future<List<ShowHive>> getDataFromFirestore() async {
+    QuerySnapshot snapshot =
+        await FirebaseFirestore.instance.collection('shows').get();
+    List<ShowHive> firestoreData = snapshot.docs.map((doc) {
+      return ShowHive.fromFirestore(doc);
+    }).toList();
+    return firestoreData;
+  }
+
+  //filling favoriteLocalBox with information received from the Firecloud
+  void favoriteLocalBoxFilling(List<ShowHive> firestoreData) async {
+    for (int i = 0; i < firestoreData.length; i++) {

```

Comment:
Have you looked into changing this for loop to the **for in**?

Reply:
Hmm.. I missed this loop. Done.